stl/inc/stacktrace

```diff
@@ -0,0 +1,365 @@
+    _NODISCARD constexpr native_handle_type native_handle() const noexcept {
+        return _Address;
+    }
+
+    _NODISCARD constexpr explicit operator bool() const noexcept {
+        return _Address != nullptr;
+    }
+
+    _NODISCARD string description() const {
+        string _Result;
+        __std_stacktrace_description(_Address, &_Result, _Stacktrace_string_fill_impl);
+        return _Result;
+    }
+
+    _NODISCARD string source_file() const {
+        string _Result;
+        __std_stacktrace_source_file(_Address, &_Result, _Stacktrace_string_fill_impl);
+        return _Result;
+    }
+
+    _NODISCARD uint_least32_t source_line() const noexcept /* strengthened */ {
+        return __std_stacktrace_source_line(_Address);
+    }
+
+    _NODISCARD_FRIEND constexpr bool operator==(const stacktrace_entry&, const stacktrace_entry&) noexcept = default;
+
+    _NODISCARD_FRIEND constexpr strong_ordering operator<=>(
+        const stacktrace_entry&, const stacktrace_entry&) noexcept = default;
+
+private:
+    void* _Address = nullptr;
+};
+
+template <class _Alloc>
+class basic_stacktrace {
+private:
+    using _Frames_t = vector<stacktrace_entry, _Alloc>;
+
+public:
+    using value_type             = stacktrace_entry;
+    using const_reference        = const value_type&;
+    using reference              = value_type&;
+    using const_iterator         = typename _Frames_t::const_iterator;
+    using iterator               = const_iterator;
+    using reverse_iterator       = _STD reverse_iterator<iterator>;
+    using const_reverse_iterator = _STD reverse_iterator<const_iterator>;
+    using difference_type        = typename _Frames_t::difference_type;
+    using size_type              = typename _Frames_t::size_type;
+    using allocator_type         = _Alloc;
+
+    // __declspec(noinline) to make the same behavior for debug and release.
+    // We force the current function to be always noinline and add its frame to skipped.
+
+    _NODISCARD __declspec(noinline) static basic_stacktrace
+        current(const allocator_type& _Al = allocator_type()) noexcept {
+        _TRY_BEGIN
+        basic_stacktrace _Result(_Internal_t{}, _Max_frames, _Al);
+        _Result._Frames.resize(__std_stacktrace_capture(1, static_cast<unsigned long>(_Max_frames),
+            reinterpret_cast<void**>(_Result._Frames.data()), &_Result._Hash));
+        return _Result;
+        _CATCH_ALL
+        return basic_stacktrace{};
+        _CATCH_END
+    }
+
+    _NODISCARD __declspec(noinline) static basic_stacktrace
+        current(size_type _Skip, const allocator_type& _Al = allocator_type()) noexcept {
+        _TRY_BEGIN
+        basic_stacktrace _Result(_Internal_t{}, _Max_frames, _Al);
+        _Result._Frames.resize(__std_stacktrace_capture(static_cast<unsigned long>(_Skip + 1),
+            static_cast<unsigned long>(_Max_frames), reinterpret_cast<void**>(_Result._Frames.data()), &_Result._Hash));
+        return _Result;
+        _CATCH_ALL

```

Comment:
please split this into two calls; this is really difficult to understand without a named return value from `__std_stacktrace_capture`:

```cxx
        unsigned short actual_size = __std_stacktrace_capture(...);
        _Result._Frames.resize(actual_size);
```

(same comment above)